lib/singleton.rb

```diff
@@ -112,40 +112,31 @@ def _dump(depth = -1)
   module SingletonClassMethods # :nodoc:
 
     def clone # :nodoc:
-      Singleton.__init__(super)
+      super.include(Singleton)
```

Comment:
I'm guessing this does not work correctly, since `@singleton__instance__` is not unset for the clone.  At the least, you would want to remove the `@singleton__instance__` instance variable from the cloned object before the method returned, if the instance variable exists.

Reply:
good catch!  it worked only because the tests cloned before calling `.instance` (so `@singleton_instance__` was nil during cloning).  I added a test to cover this